File: frontEND/src/app/entidad/factura-respuesta.ts
import { Factura } from './factura';
import { DetalleVenta } from './detalle-venta';
import { DetalleVentaConImpuestos } from './detalle-venta-con-impuestos';

export interface FacturaRespuesta {
  mensaje: string;
  factura: Factura;
  detalles: DetalleVentaConImpuestos[];
}

File: backEND/src/main/java/com/example/demo/dto/FacturaRespuesta.java
package com.example.demo.dto;

import java.util.List;
import com.example.demo.modelo.Factura;

public class FacturaRespuesta {
    private String mensaje;
    private Factura factura;
    private List<detalleVentaConImpuestos> detalles;

    // Getters y setters
    public String getMensaje() {
        return mensaje;
    }

    public void setMensaje(String mensaje) {
        this.mensaje = mensaje;
    }

    public Factura getFactura() {
        return factura;
    }

    public void setFactura(Factura factura) {
        this.factura = factura;
    }

    public List<detalleVentaConImpuestos> getDetalles() {
        return detalles;
    }

    public void setDetalles(List<detalleVentaConImpuestos> detalles) {
        this.detalles = detalles;
    }
}
